src/libPMacc/include/cuSTL/cursor/accessor/LinearInterpAccessor.hpp

```diff
@@ -0,0 +1,173 @@
+/**
+ * Copyright 2015-2016 Heiko Burau
+ *
+ * This file is part of libPMacc.
+ *
+ * libPMacc is free software: you can redistribute it and/or modify
+ * it under the terms of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * libPMacc is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with libPMacc.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "pmacc_types.hpp"
+#include "cuSTL/cursor/traits.hpp"
+#include "traits/GetInitializedInstance.hpp"
+#include "algorithms/math/defines/modf.hpp"
+#include "math/vector/Int.hpp"
+
+namespace PMacc
+{
+namespace cursor
+{
+
+/** Performs a 1D, 2D or 3D, linear interpolation on access.
+ *
+ * \tparam T_Cursor input data
+ */
+template<typename T_Cursor, int dim = cursor::traits::dim<T_Cursor>::value>
+struct LinearInterpAccessor;
+
+template<typename T_Cursor>
+struct LinearInterpAccessor<T_Cursor, DIM1>
+{
+    typedef T_Cursor Cursor;
+    typedef typename Cursor::ValueType type;
```

Comment:
`type` is a very generic name for a type, what about `CursorValueType` or `ValueType`.